### PR TITLE
Add API Token Authentication

### PR DESCRIPTION
**Pull request by jastBytes**
For certain use-cases it is necessary be able to issue long-living tokens. One use-case is to let an IAM authenticate against the SCIM endpoint of Monoskope. Therefore an authentication token needs to be issued which is long-lived.

See https://github.com/finleap-connect/monoctl/pull/14 for the client side implementation.

### CODE REVIEW

File: api/gateway/messages.proto [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+option go_package = "github.com/finleap-connect/monoskope/pkg/api/gateway";
+
+package gateway;
+
+message UpstreamAuthenticationRequest {
+  // callback_url is the URL where the authorization code
+  // will be redirected to by the upstream IDP
+  string callback_url = 1 [ (validate.rules).string.pattern =
+                                "^(https?://)?[^\\s/$.?#/_].[^\\s_]*$" ];
```
[the remark this answers is not in the document]
Comment (jastBytes) [reply api/gateway/messages.proto:32]: Fully agree! Will update.

File: api/gateway/messages.proto [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+message UpstreamAuthenticationResponse {
+  // upstream_idp_redirect is the URL of the IDP to authenticate against
+  string upstream_idp_redirect = 1;
```
[the remark this answers is not in the document]
Comment (jastBytes) [reply api/gateway/messages.proto:37]: That is the response of the server which won't be validated by the middleware.

File: api/gateway/messages.proto [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+// ClusterAuthTokenRequest is send in order to retrieve an auth token valid to
+// authenticate against a certain cluster with a specific role.
+message ClusterAuthTokenRequest {
+  // Unique identifier of the cluster (UUID 128-bit number)
+  string cluster_id = 1 [ (validate.rules).string.uuid = true ];
+  // Kubernetes role name
+  string role = 2
+      [ (validate.rules).string = {pattern : "^[a-z]+$", max_bytes : 60} ];
```
[the remark this answers is not in the document]
Comment (jastBytes) [reply api/gateway/messages.proto:106]: Adjusted!

**Comment by jastBytes** [comment]
@maresac I adjusted according to your comments.

**Review by maresac — APPROVED** [review]

**Pull request merged**